Cross-platform build broken?

<!-- Thanks for filing an issue! Before hitting the button, please answer these questions.-->

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG REPORT

<!--
If this is a BUG REPORT, please:
  - Fill in as much of the template below as you can.  If you leave out
    information, we can't help you as well.

If this is a FEATURE REQUEST, please:
  - Describe *in detail* the feature/behavior/change you'd like to see.

In both cases, be ready for followup questions, and please respond in a timely
manner.  If we can't reproduce a bug or think a feature already exists, we
might close your issue.  If we're wrong, PLEASE feel free to reopen it and
explain why.
-->

Please provide the following details:

**Environment**: Ubuntu 16.04 running in a VirtualBox VM with Mac OS X as host
latest master branch from Github
go 1.8.3

**Minikube version** (use `minikube version`): N/A
- **OS** (e.g. from /etc/os-release): Ubuntu 16.04.3 LTS (Xenial Xerus)
- **VM Driver** (e.g. `cat ~/.minikube/machines/minikube/config.json | grep DriverName`): N/A
- **ISO version** (e.g. `cat ~/.minikube/machines/minikube/config.json | grep -i ISO` or `minikube ssh cat /etc/VERSION`): N/A
- **Install tools**: N/A
- **Others**: N/A
The above can be generated in one go with the following commands (can be copied and pasted directly into your terminal):
```
minikube version
echo "";
echo "OS:";
cat /etc/os-release
echo "";
echo "VM driver": 
grep DriverName ~/.minikube/machines/minikube/config.json
```

echo "";
echo "ISO version";
grep -i ISO ~/.minikube/machines/minikube/config.json

**What happened**:
When I run the command `BUILD_IN_DOCKER=y make cross` the build starts but then fails with the following output:

```
pkg-config --cflags libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt libvirt
Package libvirt was not found in the pkg-config search path.
Perhaps you should add the directory containing `libvirt.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libvirt' found
Package libvirt was not found in the pkg-config search path.
Perhaps you should add the directory containing `libvirt.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libvirt' found
...
... many lines omitted
...
pkg-config: exit status 1
Makefile:296: recipe for target 'out/docker-machine-driver-kvm2' failed
make: *** [out/docker-machine-driver-kvm2] Error 2
```


**What you expected to happen**: the build completes successfully with all binaries located in /out

**How to reproduce it** (as minimally and precisely as possible): follow the instructions in https://github.com/kubernetes/minikube/blob/master/docs/contributors/build_guide.md and then in https://github.com/kubernetes/minikube/blob/master/docs/contributors/releasing_minikube.md

**Output of `minikube logs` (if applicable)**: N/A

**Anything else do we need to know**:

--- a/Makefile
+++ b/Makefile
@@ -115,7 +115,7 @@ else
 endif
 
 .PHONY: e2e-%-amd64
-e2e-%-amd64:
+e2e-%-amd64: out/minikube-%-amd64
 	GOOS=$* GOARCH=amd64 go test -c k8s.io/minikube/test/integration --tags="$(MINIKUBE_INTEGRATION_BUILD_TAGS)" -o out/$@
 
 e2e-windows-amd64.exe: e2e-windows-amd64
@@ -153,6 +153,18 @@ test-iso:
 .PHONY: test-pkg
 test-pkg/%:
 	go test -v -test.timeout=30m $(REPOPATH)/$* --tags="$(MINIKUBE_BUILD_TAGS)"
+
+.PHONY: all
+all: cross drivers e2e-cross images
+
+.PHONY: drivers
+drivers: out/docker-machine-driver-hyperkit out/docker-machine-driver-kvm2
+
+.PHONY: images
+images: localkube-image localkube-dind-image localkube-dind-image-devshell
+	gcloud docker -- push gcr.io/k8s-minikube/localkube-image:$(TAG)
+	gcloud docker -- push gcr.io/k8s-minikube/localkube-dind-image:$(TAG)
+	gcloud docker -- push gcr.io/k8s-minikube/localkube-dind-image-devshell:$(TAG)
 
 .PHONY: integration
 integration: out/minikube

--- a/Makefile
+++ b/Makefile
@@ -183,7 +183,7 @@ $(GOPATH)/bin/go-bindata:
 	GOBIN=$(GOPATH)/bin go get github.com/jteeuwen/go-bindata/...
 
 .PHONY: cross
-cross: out/localkube out/minikube-linux-amd64 out/minikube-darwin-amd64 out/minikube-windows-amd64.exe out/docker-machine-driver-hyperkit out/docker-machine-driver-kvm2
+cross: out/localkube out/minikube-linux-amd64 out/minikube-darwin-amd64 out/minikube-windows-amd64.exe
 
 .PHONY: e2e-cross
 e2e-cross: e2e-linux-amd64 e2e-darwin-amd64 e2e-windows-amd64.exe

--- a/hack/jenkins/minikube_cross_build_and_upload.sh
+++ b/hack/jenkins/minikube_cross_build_and_upload.sh
@@ -23,8 +23,14 @@
 
 set -e
 
+export BUILD_IN_DOCKER=y
+export TAG=$ghprbActualCommit
+export GOPATH=/var/lib/jenkins/go
+
+make -j 16 all
+
 # Clean up exited containers
-docker rm $(docker ps -q -f status=exited) || true
+docker kill $(docker ps -q) || true
 
 gsutil cp gs://minikube-builds/logs/index.html gs://minikube-builds/logs/${ghprbPullId}/index.html
 
@@ -38,31 +44,10 @@ if out="$(git diff ${ghprbActualCommit} --name-only $(git merge-base origin/mast
 	make release-iso
 fi
 
-export GOPATH=~/go
-
-# Build the e2e test target for Darwin and Linux. We don't run tests on Windows yet.
-# We build these on Linux, but run the tests on different platforms.
-# This makes it easier to provision slaves, since they don't need to have a go toolchain.'
-# Cross also builds the hyperkit and kvm2 drivers.
-BUILD_IN_DOCKER=y make cross 
-BUILD_IN_DOCKER=y make e2e-cross
 cp -r test/integration/testdata out/
-
-BUILD_IN_DOCKER=y make out/docker-machine-driver-hyperkit
-BUILD_IN_DOCKER=y make out/docker-machine-driver-kvm2
 
 # Don't upload the buildroot artifacts if they exist
 rm -r out/buildroot || true
 
 # Upload everything we built to Cloud Storage.
 gsutil -m cp -r out/* gs://minikube-builds/${ghprbPullId}/
-
-# Upload containers for the PR:
-TAG=$ghprbActualCommit make localkube-image
-gcloud docker -- push gcr.io/k8s-minikube/localkube-image:$ghprbActualCommit
-
-TAG=$ghprbActualCommit make localkube-dind-image
-gcloud docker -- push gcr.io/k8s-minikube/localkube-dind-image:$ghprbActualCommit
-
-TAG=$ghprbActualCommit make localkube-dind-image-devshell
-gcloud docker -- push gcr.io/k8s-minikube/localkube-dind-image-devshell:$ghprbActualCommit

--- a/hack/jenkins/minikube_cross_build_and_upload.sh
+++ b/hack/jenkins/minikube_cross_build_and_upload.sh
@@ -44,8 +44,12 @@ export GOPATH=~/go
 # We build these on Linux, but run the tests on different platforms.
 # This makes it easier to provision slaves, since they don't need to have a go toolchain.'
 # Cross also builds the hyperkit and kvm2 drivers.
-BUILD_IN_DOCKER=y make cross e2e-cross
+BUILD_IN_DOCKER=y make cross 
+BUILD_IN_DOCKER=y make e2e-cross
 cp -r test/integration/testdata out/
+
+BUILD_IN_DOCKER=y make out/docker-machine-driver-hyperkit
+BUILD_IN_DOCKER=y make out/docker-machine-driver-kvm2
 
 # Don't upload the buildroot artifacts if they exist
 rm -r out/buildroot || true

